### DISCUSSION TITLE
5.15.57编译的固件出现了了NAT环回的问题

### DISCUSSION THREAD

**Discussion by eightsheep67** [coolsnowwolf/lede#9851] [Bugs / 问题反馈]
新编译了固件，但是出现了很奇怪的问题。
我的设置是这样：
防火墙端口1111指向主路由，192.168.1.1
防火墙端口2222指向NAS，192.168.1.100
域名A我指定主机名IP为192.168.1.1
域名B我指定主机名IP为192.168.1.100
两个域名我都DDNS到我家的公网IP了。
正常情况下，我在外网不管访问域名A还是域名B，端口1111就会到达主路由，端口2222就会到达NAS
然后在内网NAT环回下，因为一个域名只能绑定一个IP，所以我访问域名A:1111，可以到达主路由，访问域名B:2222就会到达NAS。
但是新固件下，仅仅只是我家的几台IPHONE啊，访问域名B:2222到达NAS没问题，但是访问域名A:1111就出现无法连接，奇怪的是访问域名A:2222居然也可以到达NAS。仿佛在iphone上，openwrt上的主机名设定就失效了？openwrt都认为我是从外网访问？我电脑上的NAT回环都很正常。域名A:1111正常，而且域名A:2222打不开的也是正常的。
之前的固件没有这种情况？而且怎么只对iphone产生这种效果，这个实在无语……

**Comment by eightsheep67** [comment]
我把主机名设定都删了，就都正常了……

**Comment by eightsheep67** [comment]
可是删了主机名，就会导致未联网的时候无法通过域名访问家里的路由器和nas……有什么方法排查一下吗

**Comment by zzy19868** [comment]
相同的config，我在本地编译就会有环回问题，在线编译的就没有。估计还是网络环境的问题。
